src/cdk/datetime/datepicker-input.ts

```diff
@@ -0,0 +1,324 @@
+import {CdkDatepicker} from './datepicker';
+
+
+/**
+ * Provider that allows the datepicker to register as a ControlValueAccessor.
+ */
+export const DATEPICKER_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => CdkDatepickerInput),
+  multi: true
+};
+
+
+/**
+ * Provider that allows the datepicker to register as a ControlValueAccessor.
+ */
+export const DATEPICKER_VALIDATORS: any = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => CdkDatepickerInput),
+  multi: true
+};
+
+
+/**
+ * An event used for datepicker input and change events. For consistency, we always use
+ * DatepickerInputEvent instead.
+ */
+export class DatepickerInputEvent<D> {
+  /** The new value for the target datepicker input. */
+  value: D | null;
+
+  constructor(
+    /** Reference to the datepicker input component that emitted the event. */
+    public target: CdkDatepickerInput<D>,
+      /** Reference to the native input element associated with the datepicker input. */
+      public targetElement: HTMLElement) {
+    this.value = this.target.value;
+  }
+}
+
+
+/** Directive used to connect an input to a CdkDatepicker. */
+@Directive({
+  selector: 'input[cdkDatepicker]',
+  providers: [
+    DATEPICKER_VALUE_ACCESSOR,
+    DATEPICKER_VALIDATORS,
+  ],
+  host: {
+    '[attr.aria-owns]': '(_datepicker.id) || null',
+    '[attr.min]': 'min ? _dateAdapter.toIso8601(min) : null',
+    '[attr.max]': 'max ? _dateAdapter.toIso8601(max) : null',
+    '[disabled]': 'disabled',
+    '(change)': '_onChange()',
+    '(blur)': '_onBlur()',
+  },
+  exportAs: 'cdkDatepickerInput',
+})
+export class CdkDatepickerInput<D> implements AfterContentInit, ControlValueAccessor, OnDestroy,
+  Validator {
+  /** The datepicker that this input is associated with. */
+  @Input('cdkDatepicker')
+  set datepicker(value: CdkDatepicker<D>) {
+    this.registerDatepicker(value);
+  }
+  _datepicker: CdkDatepicker<D>;
+
+  /** Registering datepicker with this input. */
+  private registerDatepicker(value: CdkDatepicker<D>) {
+    if (value) {
+      this._datepicker = value;
+      this._datepicker._registerInput(this);
+    }
+  }
+
+  /** Function that can be used to filter out dates within the datepicker. */
+  @Input()
+  set datepickerFilter(value: (date: D | null) => boolean) {
+    this._dateFilter = value;
+    this._validatorOnChange();
+  }
+  _dateFilter: (date: D | null) => boolean;
+
+  /** The value of the input. */
+  @Input()
+  get value(): D | null { return this._value; }
+  set value(value: D | null) {
+    value = this._dateAdapter.deserialize(value);
+    this._lastValueValid = !value || this._dateAdapter.isValid(value);
+    value = this._getValidDateOrNull(value);
+    const oldDate = this.value;
+    this._value = value;
+    this.emitValue(oldDate, value);
+  }
+  private _value: D | null;
+
+  /** Emitting change detection on the value. */
+  emitValue(oldDate: D | null, value: D | null) {
```

Comment:
can you move this down with the other methods? it's weird that it shows up in the middle of the `@Input` section

Reply:
Also, lets call it `emitValueIfChanged(oldValue, newValue)`